test/functional/spec-runner/context.js

```diff
@@ -30,8 +30,18 @@ class Thread {
 }
 
 class TestRunnerContext {
-  constructor() {
-    this.url = null;
+  constructor(opts) {
+    const d = {
```

Comment:
Nit: I try to avoid single letter variable names to improve readability, could we change this to `defaults`?

Reply:
good rule 👍